app/src/main/java/com/aleksandra/go4mytrip/PackingListRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.aleksandra.go4mytrip
+
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+
+open class PackingListRepository {
+
+    private lateinit var uid: String
+
+    fun getDatabaseReferencePackingList() : DatabaseReference{
```

Comment:
the repository should be responsible for loading data, so if you used for instance RxJava, you could write it as:
fun getPackingList() : Single<List<PackingModel>>

you don't have RxJava or Kotlin coroutines in your project, so you can define and pass a listener to getPackingList

the idea was to hide DatabaseReference inside the repository

Reply:
https://proandroiddev.com/the-real-repository-pattern-in-android-efba8662b754